### ISSUE TITLE
EPIC - CLI : ps-upgrade:upgrade - implement upgrade files

### ISSUE THREAD

**Issue by fabienvallon** [PrestaShop/PrestaShop#35169]
As a user I want to upgrade prestashop so I can have an up2date software  

This is part of [CLI : ps-upgrade:upgrade](https://github.com/orgs/PrestaShop/projects/7/views/43?layout=board&pane=issue&itemId=44108684) 

upgrade files is a step of the upgrade 

## Description 

**\<FILE_ORIGINAL\>** : a file (ex: a php file ) of what the prestashop installed should have (extract from the xml file , example from prestashop-8.1.3.xml if prestashop-8.1.3 is currently installed )

**\<FILE_CURRENT\>** : a file (ex: a php file ) of prestashop **currently installed (real)**  

**\<FILE_TARGET\>** :  a file (ex: a php file ) of  prestashop  **to install** should have (extract from the xml file , example from prestashop-8.1.3.xml if prestashop-8.1.3 is currently installed )

**\<MD5_FILE_ORIGINAL\>** : md5sum of \<FILE_ORIGINAL\> extract form xml file

**\<MD5_FILE_CURRENT\>** : md5sum of \<FILE_CURRENT\> , compute from with the current installation (at --prestashop-path \<PATH\> ) 

**\<MD5_FILE_TARGET\>** : md5sum of \<FILE_TARGET\> extract form xml file ( prestashop to install ) 

xml file (SOURCE & TARGET xml file ) should be download if not available on the disk / filesystem ( see TS ) 

--- 

####  if \<MD5_FILE_TARGET\> == \<MD5_FILE_ORIGINAL\> 

Then do nothing ( means let the file as is ) 

#### if \<MD5_FILE_TARGET\> ( or \<FILE_TARGET\> ? ) does not exist 

Then delete \<FILE_CURRENT\>

----

#### . if  \<MD5_CURRENT\> == \<MD5_TARGET\>

Does nothing

#### . if  \<MD5_CURRENT\> != \<MD5_TARGET\>

Suivant ce qui a été vu en check 

Message : 
You have modified  \<FILE_CURRENT> of your current installation : what do you want to do here ? 
[1]  Backup it  & Replace it by our new implementation 
[2] Replace it by our new implementation without backup ( WARNING you will loose your modifications ) 
[3]  Keep your modified file ? ( WARNING this could make your new installation unusable ) 

---- 

( revoir messages Fabien / @Julievrz  ? ) 

#### if it does not exists into current :  \<FILE_CURRENT\> 

Then add it 

#### if it exists into current \<FILE_CURRENT\> 

Message : 
```
You have a  <FILE_CURRENT> in your current installation : what to do here ? 
[1]  Backup it  & Replace it by our new implementation 
[2] Replace it by our new implementation without backup ( WARNING you will loose your modifications ) 
[3]  Keep your modified file ? ( WARNING this could make your new installation unusable )
```

### if ORIGINAL_PACKAGE.xml is not available 

1 . Download xml file for \<ORIGINAL PACKAGE\>

Message: 
```
Dowloading <ORIGINAL_PACKAGE.xml> file .....<STATUS>
```

if download is ok 
\<STATUS\> = OK

if download is ko 
\<STATUS\> = KO + Message \<ERROR\>

2. Download ask md5 check for xml file for \<SOURCE PACKAGE\> through API 

Verify integrity for \<SOURCE_PACKAGE.xml.M\> file .....\<STATUS\>

if md5sum is ok 
\<STATUS\> = OK

if md5sum is ko 
\<STATUS\> = KO: integrity check failed 

### Acceptance test : OK  - Without Download

``` 
Upgrading files ....OK 
``` 

### Acceptance test : KO  - Without Download

``` 
Upgrading files ....KO <ERRORS>  
``` 
See with dev for  \<ERRORS\>  

### Acceptance test : OK  - With Download

``` 
Dowloading <ORIGINAL_PACKAGE.xml> file ....OK
Upgrading files ....OK 
``` 

### Acceptance test : KO  - With Download KO

``` 
Dowloading <ORIGINAL_PACKAGE.xml> file ....KO
``` 
( TODO ) 

### Technical specs

- [ ] #35159 
- [ ] CliApp: Create this step in `ps-upgrade:shop:upgrade` command.

**Comment by Hlavtox** [comment]
Ping @Quetzacoalt91 Thomas, can I close this?

**Comment by Quetzacoalt91** [comment]
Yep, this has been replaced by Update Assistant 7 and modals about missing / altered files.
